tcms/management/tests/test_baseurl.py

```diff
@@ -0,0 +1,28 @@
+from io import StringIO
+
+from django.core.management import call_command
+from django.test import TestCase
+
+from tcms.tests.factories import SiteFactory
+
+
+class TestBaseurl(TestCase):
+    """Test manage.py baseurl command"""
+
+    def test_check_current_url(self):
+        """Test command without arguments returns current Base URL"""
+        site = SiteFactory()
+        out = StringIO()
+        call_command('baseurl', stdout=out)
+        self.assertEqual(
+            'URL: "%s", name: "%s"\n' % (site.domain, site.name),
+            out.getvalue())
+
+    def test_set_url(self):
+        """Test if command sets Base URL correctly"""
+        out = StringIO()
+        newurl = "https://kiwi.test.bogus:1234"
+        call_command('baseurl', newurl, stdout=out)
+        site = SiteFactory()
+        self.assertEqual(newurl, site.domain)
```

Comment:
Where is this variable `site` coming from ?

Reply:
It comes from the SiteFactory(). Since the record exists the factory returns its values. I use this to check what is in the database.

Reply:
I see now. That is completely wrong. You have to set the new domain name and then get the object from the database and assert that its domain is as expected. These are direct operations, they don't need a proxy between them.